packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -342,13 +342,29 @@ export default class Proxy {
       xoaUpgradeTimeout
     )
 
-    await this.checkProxyHealth(proxyId)
+    const { success, error } = await this.checkProxyHealth(proxyId)
+    if (!success) {
+      throw new Error(error.message)
```

Comment:
We are loosing the original code, stack trace and other error properties here.

Reply:
Please see https://github.com/vatesfr/xen-orchestra/pull/5191#discussion_r470676690

Reply:
The hide of other properties is for security reason. Please see https://github.com/vatesfr/xen-orchestra/pull/5191#discussion_r470676690.

Reply:
Unrelated, I'm talking about the API method `proxy.checkHealth` here.

Reply:
Yes, but it's the same thing. If we are afraid that `checkProxyHealth` error contains sensitive data, we need to hide props in other  `checkProxyHealth` calls. It's not a specific issue but it's a general issue.

Reply:
No, we don't want to hide anything inside `xo-server` as it would impede debugging.

We should only be careful via the API and `xo-server` already takes care of any errors that are not meant to cross the API by replacing them with *unknown error from the peer*.